libs/vkd3d/state.c

```diff
@@ -4158,29 +4190,30 @@ VkPipeline d3d12_pipeline_state_create_pipeline_variant(struct d3d12_pipeline_st
             rtv_formats[i] = VK_FORMAT_UNDEFINED;
     }
 
+    memset(&pipeline_desc, 0, sizeof(pipeline_desc));
     pipeline_desc.sType = VK_STRUCTURE_TYPE_GRAPHICS_PIPELINE_CREATE_INFO;
     pipeline_desc.pNext = &rendering_info;
-    pipeline_desc.flags = 0;
     pipeline_desc.stageCount = graphics->stage_count;
     pipeline_desc.pStages = graphics->stages;
-    pipeline_desc.pVertexInputState = &input_desc;
-    pipeline_desc.pInputAssemblyState = &ia_desc;
-    pipeline_desc.pTessellationState = &tessellation_info;
     pipeline_desc.pViewportState = &vp_desc;
     pipeline_desc.pRasterizationState = &graphics->rs_desc;
     pipeline_desc.pMultisampleState = &graphics->ms_desc;
     pipeline_desc.pDepthStencilState = &graphics->ds_desc;
     pipeline_desc.pColorBlendState = &graphics->blend_desc;
     pipeline_desc.pDynamicState = &dynamic_create_info;
     pipeline_desc.layout = graphics->pipeline_layout;
-    pipeline_desc.renderPass = VK_NULL_HANDLE;
-    pipeline_desc.subpass = 0;
-    pipeline_desc.basePipelineHandle = VK_NULL_HANDLE;
     pipeline_desc.basePipelineIndex = -1;
 
     if (d3d12_device_supports_variable_shading_rate_tier_2(device))
         pipeline_desc.flags |= VK_PIPELINE_CREATE_RENDERING_FRAGMENT_SHADING_RATE_ATTACHMENT_BIT_KHR;
 
+    if (!(graphics->stage_flags & VK_SHADER_STAGE_MESH_BIT_EXT))
```

Comment:
We probably have to avoid some dynamic state in `d3d12_graphics_pipeline_state_init_dynamic_state` as well such as `VERTEX_BUFFER_STRIDE` and `TOPOLOGY`.

Reply:
`d3d12_pipeline_state_get_pipeline` can call `d3d12_pipeline_state_can_use_dynamic_stride` as well. I would assume that VBO binding count must be 0 for mesh PSOs?